### PR TITLE
Implement Nibble Data-Converter Option

### LINKED ISSUE #2055: Add Nibble (Nybl) to Data Unit Conversion
**Problem Statement**

The Nibble is not a data unit that can be converted on/in the Windows Calculator.

**Evidence or User Insights**

Nibble is often used in architecture/Networking, so it would be useful to implement it into the data conversion calculator.-->

**Proposal**

To implement an option in the Data Converter section to convert Nibble to other units would be highly useful to users.  

**Goals**

- Add Nibble to the list of data conversions

**Non-Goals**

**Low-Fidelity Concept**

1 Nibble is 4 bits or half a byte.  1 byte is 2 nibbles, 1 bit is .25 nibble 

**Requested Assignment**
I can implement this. I have already written the code for this. It is relatively easy to write/implement.

### PR DESCRIPTION
**Pull request by AlPasDev**
### Description of the changes:
- Implements Nibble as a data-conversion option
- Nibble can be converted to other data types

### How changes were validated:

- Added Unit Testing

### CODE REVIEW

**Comment by AlPasDev** [comment]
@microsoft-github-policy-service agree

**Comment by AlPasDev** [comment]
@mcooley @tian-lt @volkcor Please review this PR.
Thank You, Alex Pascal
alpascal@outlook.com

**Comment by AlPasDev** [comment]
@grochocki  @mcooley 
Hello, sorry to bother. Is there any way for me to get this pull request reviewed? Am I missing something, or not?
Thank you, A. Pascal

**Comment by AlPasDev** [comment]
@tian-lt @volkcor #2056

**Comment by tian-lt** [comment]
Hi @AlPasDev, sorry for my late response and thank you for proposing this PR.
Let me firstly trigger the pipeline to verify some basic criteria.

File: src/Calculator/Resources/en-GB/Resources.resw [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (tian-lt) [inline src/Calculator/Resources/en-GB/Resources.resw:1]: Added or removed a UTF8-BOM header unintentionally? If yes, please revert the change for this line.

Comment (AlPasDev) [reply src/Calculator/Resources/en-GB/Resources.resw:1]: Yes it was unintentional.  @tian-lt

Comment (AlPasDev) [reply src/Calculator/Resources/en-GB/Resources.resw:1]: @tian-lt How do I revert the single line?

Comment (AlPasDev) [reply src/Calculator/Resources/en-GB/Resources.resw:1]: @tian-lt , it wasn't removed or added. It was just moved based on the addition of my code.

Comment (tian-lt) [reply src/Calculator/Resources/en-GB/Resources.resw:1]: > Yes it was unintentional. @tian-lt How do I revert the single line?

You can simply use the same file from the main branch to replace this one and then commit the change.

File: src/CalcViewModel/DataLoaders/UnitConverterDataConstants.h [hunk 2]
```diff
@@ -165,6 +165,7 @@ namespace CalculatorApp
             Data_Zetabytes = UnitStart + 164,
             Area_Pyeong = UnitStart + 165,
             Energy_Kilowatthour = UnitStart + 166,
+			Data_Nibble = UnitStart + 167,
             UnitEnd = Energy_Kilowatthour
```
Comment (tian-lt) [inline src/CalcViewModel/DataLoaders/UnitConverterDataConstants.h:169]: `UnitEnd` needs to be updated to `Data_Nibble`. i.e. `UnitEnd = Data_Nibble`.

File: src/Calculator/Resources/en-GB/Resources.resw (hunk 1 above)
Comment (AlPasDev) [reply src/Calculator/Resources/en-GB/Resources.resw:1]: @tian-lt fixed this.

Comment (tian-lt) [reply src/Calculator/Resources/en-GB/Resources.resw:1]: Looks like the irrelevant changes increased a lot.
Did you copy the `en-US` to `en-GB`? (I saw the word _Colour_ has been changed into _Color_.

Comment (AlPasDev) [reply src/Calculator/Resources/en-GB/Resources.resw:1]: I did accidentally copy the wrong file.. my mistake. @tian-lt 
I don't know why that BOM is being added. I didn't touch that line

Comment (tian-lt) [reply src/Calculator/Resources/en-GB/Resources.resw]: No worries. I think we are almost good to finish this PR.

One last **TODO**: Revert all the changes made to this en-GB resource file. It is unnecessary to add localization strings to `Resources.resw` files except the ***en-US*** one, because we have a team dedicated to translating en-US strings to all the other languages Calculator supports.

Comment (AlPasDev) [reply src/Calculator/Resources/en-GB/Resources.resw]: @tian-lt Finished

**Review by tian-lt — APPROVED** [review]

**Review by tian-lt — CHANGES_REQUESTED** [review]
Found another problem: the unit order in UI doesn't make sense. Nibble should be place between `Bits` and `Bytes`, but in fact, it is placed at the last position (behind the `Yobibytes`:
image

**Comment by AlPasDev** [comment]
@tian-lt fixed up the issue you requested.

**Comment by AlPasDev** [comment]
@Tian-lt Please let me know if it's all good.

**Review by tian-lt — APPROVED** [review]

**Comment by tian-lt** [comment]
> @tian-lt Please let me know if it's all good.

@AlPasDev the changes look good to me. I merged this PR into the main branch. Some release tests will be schedule for the next feel weeks. Hopefully we can see the feature rolls out in next few months.
